logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -382,40 +381,36 @@
       after(:each) { File.unlink(new_file) }
 
       it "resets the pipeline metric collector" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:events][:in].value
+        value = Witness.instance.pipeline("main").events.snitch.in
         expect(value).to be <= new_config_generator_counter
       end
 
       it "does not reset the global event count" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/events")[:stats][:events][:in].value
+        value = Witness.instance.events.snitch.in
         expect(value).to be > initial_generator_threshold
       end
 
       it "increases the successful reload count" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:successes].value
+        value = Witness.instance.pipeline("main").reloads.snitch.successes
         expect(value).to eq(1)
-        instance_value = snapshot.metric_store.get_with_path("/stats")[:stats][:reloads][:successes].value
+        instance_value = Witness.instance.reloads.snitch.successes
         expect(instance_value).to eq(1)
       end
 
       it "does not set the failure reload timestamp" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:last_failure_timestamp].value
+        value = Witness.instance.pipeline("main").reloads.snitch.last_failure_timestamp
         expect(value).to be(nil)
       end
 
       it "sets the success reload timestamp" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:last_success_timestamp].value
+        value = Witness.instance.pipeline("main").reloads.snitch.last_success_timestamp
         expect(value).to be_a(Timestamp)
       end
 
       it "does not set the last reload error" do
-        snapshot = subject.metric.collector.snapshot_metric
-        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:last_error].value
```

Comment:
Note - the Ruby implementation used a Ruby Hash here. The `Witness` data model splits these up into explicit `message` and `backtrace` values.